make/photon/prepare/utils/configs.py

```diff
@@ -372,24 +361,38 @@ def parse_yaml_config(config_file_path, with_notary, with_clair, with_trivy, wit
 
 
 def get_redis_url(db, redis=None):
-    """Returns redis url with format `redis://[arbitrary_username:password@]ipaddress:port/database_index`
+    """Returns redis url with format `redis://[arbitrary_username:password@]ipaddress:port/database_index?idle_timeout_seconds=30`
 
     >>> get_redis_url(1)
     'redis://redis:6379/1'
-    >>> get_redis_url(1, {'host': 'localhost', 'password': 'password'})
+    >>> get_redis_url(1, {'host': 'localhost:6379', 'password': 'password'})
     'redis://anonymous:password@localhost:6379/1'
+    >>> get_redis_url(1, {'host':'host1:26379,host2:26379', 'sentinel_master_set':'mymaster', 'password':'password1'})
+    'redis+sentinel://anonymous:password@host1:26379,host2:26379/mymaster/1'
+    >>> get_redis_url(1, {'host':'host1:26379,host2:26379', 'sentinel_master_set':'mymaster', 'password':'password1','idle_timeout_seconds':30})
+    'redis+sentinel://anonymous:password@host1:26379,host2:26379/mymaster/1?idle_timeout_seconds=30'
+
     """
     kwargs = {
-        'host': 'redis',
-        'port': 6379,
+        'host': 'redis:6379',
         'password': '',
     }
     kwargs.update(redis or {})
-    kwargs['db'] = db
+    kwargs['scheme'] = kwargs.get('sentinel_master_set', None) and 'redis+sentinel' or 'redis'
+    kwargs['db_part'] = db and ("/%s" % db) or ""
+    kwargs['sentinel_part'] = kwargs.get('sentinel_master_set', None) and ("/" + kwargs['sentinel_master_set']) or ''
+    kwargs['password_part'] = kwargs.get('password', None) and (':%s@' % kwargs['password']) or ''
 
-    if kwargs['password']:
-        return "redis://anonymous:{password}@{host}:{port}/{db}".format(**kwargs)
-    return "redis://{host}:{port}/{db}".format(**kwargs)
+    return "{scheme}://{password_part}{host}{sentinel_part}{db_part}".format(**kwargs) + get_redis_url_param(kwargs)
```

Comment:
recommend only store content in variables like 'db_part', 'sentinel_part', 'password_part'. And the delimiter like`/` ,`@` can rendered in format string.